beps/0001-notifications-system/README.md

```diff
@@ -0,0 +1,255 @@
+
+### Architecture Overview
+
+The diagram below show an overview of the notifications system architecture and the interaction between the different components.
+
+![notifications system architecture diagram](./notifications-architecture.drawio.svg)
+
+The components that are presented with dashed lines are not directly part of this proposal, but included to be considered as future extensions.
+
+Let's walk through the process of sending of a notification to a user:
+
+1. A backend plugin that wants to send a notification to a user uses the `NotificationService`, which in turn makes a POST request to the notification backend. The request contains a filter to select the target users, as well as the notification payload.
+
+   1b. As the notification backend handles the request it could optionally invoke a series of notification processors. These processors can be used to transform the notification payload, send the notification through other channels, and decide whether the regular notification flow should continue or not. This step is not in scope for this proposal.
+
+2. The notification backend stores the received notification in its database. At this point the notification is available from the notification backend's read API.
+3. The notification backend published an event to the event bus on the `'signal'` topic. This event payload contains the signaling channel, in this case `'notifications'`, as well as the target user ID and the notification ID, but not the notification payload.
+4. Each instance of the signal backend plugin subscribes to the `'signal'` topic and receives the event.
+5. Each signal backend instance has a set of push channels set up to online users. The incoming event is filtered based on the target user ID and the signaling is pushed through all connections with a matching user ID.
+6. The signal frontend plugin receives the signal and forwards it to all subscribers of the target channel. In this case, the notifications frontend plugin, who is subscribed to the `'notifications'` channel.
+7. The notifications frontend plugin receives the signal with the notification ID. It then makes a request to the notification backend to fetch the notification payload and then displays the notification to the user.
+
+   7b. Rather than rendering the notification directly, the notifications frontend plugin leaves that to an extension installed by the frontend part of the backend plugin that sent the notification. This allows for a more customized notification experience which can also contain custom actions and data display. This step is not in scope for this proposal.
+
+## Design Details
```

Comment:
Maybe detail out also the database schema as it explains a lot about how the notifications should look like in all parts of the plugin?

Reply:
Yep! The intention is for is all to fill out this section with more details in followup PRs